src/main/java/seedu/moneygowhere/logic/parser/AddReminderCommandParser.java

```diff
@@ -30,16 +32,20 @@ public AddReminderCommand parse(String args) throws ParseException {
                 || !argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddReminderCommand.MESSAGE_USAGE));
         }
-        if (argMultimap.getValue(PREFIX_MESSAGE).isPresent()) {
-            String reminderMessage = argMultimap.getValue(PREFIX_MESSAGE).get().trim();
 
-            if (reminderMessage.isEmpty()) {
-                throw new ParseException(ReminderMessage.MESSAGE_CONSTRAINTS);
-            }
+        String reminderMessage = argMultimap.getValue(PREFIX_MESSAGE).get().trim();
+
+        if (reminderMessage.isEmpty()) {
+            throw new ParseException(ReminderMessage.MESSAGE_CONSTRAINTS);
         }
 
         Date date = ParserUtil.parseDate(argMultimap.getValue(PREFIX_DATE).get());
-        ReminderMessage message = ParserUtil.parseMessage(argMultimap.getValue(PREFIX_MESSAGE).get());
+        ReminderMessage message = ParserUtil.parseMessage(reminderMessage);
+
+        if (date.dateValue.isAfter(LocalDate.now())
+                && date.dateValue.getYear() - LocalDate.now().getYear() > ParserUtil.DATE_TOO_FAR_RANGE) {
```

Comment:
We could probably also limit this to past reminders. Reminders 5 years ago should not be added.

Reply:
I guess we can. May be the spending as well?

Reply:
Good consideration, can create an issue